PacificPower/readPP.js

```diff
@@ -70,7 +70,98 @@ let continueVarMonthly = 0;
 let page = "";
 let pp_meter_id = "";
 
+let pp_meters_exclusion_list = null;
+let pp_meters_exclude = [];
+let pp_meters_include = [];
+let pp_meters_exclude_not_found = [];
+
+function compareMeterAgainstExclusionList(PPTable) {
+  const meter = pp_meters_exclusion_list.find(
+    (meter) => meter.pp_meter_id === PPTable.pp_meter_id,
+  );
+
+  if (!meter) {
+    console.log(
+      `Meter ${PPTable.pp_meter_id} is not in the exclusion list: NEW METER`,
+    );
+    pp_meters_exclude_not_found.push(PPTable.pp_meter_id);
+    PPArray.push(PPTable);
+    return;
+  }
+
+  switch (meter.status) {
+    case "exclude":
+      console.log(`Meter ${PPTable.pp_meter_id} is excluded from db`);
+      pp_meters_exclude.push(PPTable.pp_meter_id);
+      break;
+    case "include":
+      console.log(`Meter ${PPTable.pp_meter_id} is included in db`);
+      pp_meters_include.push(PPTable.pp_meter_id);
+      PPArray.push(PPTable);
+      break;
+    case "new":
+      console.log(
+        `Meter ${PPTable.pp_meter_id} status needs updating, include in db for now.`,
+      );
+      pp_meters_include.push(PPTable.pp_meter_id);
+      PPArray.push(PPTable);
+      break;
+    default:
+      console.log(`Meter ${PPTable.pp_meter_id} unrecognized status`);
+  }
+}
+
+async function addNewMetersToDatabase() {
+  for (let i = 0; i < pp_meters_exclude_not_found.length; i++) {
+    await axios({
+      method: "post",
+      url: `${process.env.DASHBOARD_API}/ppupload`,
+      data: {
+        id: pp_meters_exclude_not_found[i],
+        pwd: process.env.API_PWD,
+      },
+    })
+      .then((res) => {
+        console.log(
+          `RESPONSE: ${res.status}, TEXT: ${res.statusText}, DATA: ${res.data}`,
```

Comment:
I think we should do a JSON.stringify() on the res.data here, if we intend to log it, as right now nothing is actually being logged from upload output

![image](https://github.com/OSU-Sustainability-Office/automated-jobs/assets/82061589/b7c5ef9c-827d-4677-a6f4-461bb84f05f6)

Reply:
I've removed the data log altogether as it was just the reply from the DB (`DATA: {"fieldCount":0,"affectedRows":1,"insertId":9,"serverStatus":2,"warningCount":1,"message":"","protocol41":true,"changedRows":0}`). Instead, if the response is successful it will log the meter ID and that it was uploaded:

```
RESPONSE: 200, TEXT: OK
74263851 uploaded to database.
```

Reply:
LGTM